### PR TITLE
Add special widgets for realTypes

### PR DESCRIPTION
**Pull request by gselzer**
This is much easier to do than adding support for ctypes, and allows us to support interfaces as well.

### CODE REVIEW

**Comment by ctrueden** [comment]
Nice!

File: src/napari_imagej/types/widget_mappings.py [hunk 1]
```diff
@@ -44,10 +45,24 @@ def preferred_widget_for(
...
+@_widget_preference
+def _numeric_type_preference(
+    item: "jc.ModuleItem", type_hint: Union[type, str]
+) -> Optional[Union[type, str]]:
+    # We only care about mutable outputs
+    if item.isInput() and not item.isOutput():
+        # This is the best place for all RealTypes EXCEPT BooleanType
+        if issubclass(item.getType(), jc.BooleanType):
+            # Use some sort of checkbox instead
+            return None
```
Comment (gselzer) [inline src/napari_imagej/types/widget_mappings.py:57]: Does `BooleanType` actually work? Can we test this?

File: src/napari_imagej/widgets/parameter_widgets.py [hunk 2]
```diff
@@ -3,14 +3,71 @@
...
+        cls = jc.DoubleType.class_
+    elif cls == jc.IntegerType:
+        cls = jc.LongType.class_
+
+    # Return a SpinBox implementation for integer types
+    if issubclass(cls, jc.IntegerType):
+
+        class Widget(SpinBox):
+            def __init__(self, **kwargs):
+                min_val = max(cls.newInstance().getMinValue(), -(2**31))
```
Comment (gselzer) [inline src/napari_imagej/widgets/parameter_widgets.py:38]: Might want to add a comment here as well.

File: src/napari_imagej/widgets/parameter_widgets.py [hunk 3]
```diff
@@ -3,14 +3,71 @@
...
+    # Set sensible defaults for interfaces
+    if cls == jc.RealType or cls == jc.NumericType:
+        cls = jc.DoubleType.class_
+    elif cls == jc.IntegerType:
+        cls = jc.LongType.class_
+
+    # Return a SpinBox implementation for integer types
+    if issubclass(cls, jc.IntegerType):
+
+        class Widget(SpinBox):
```
Comment (gselzer) [inline src/napari_imagej/widgets/parameter_widgets.py:36]: We could theoretically use dynamic inheritance to make only one class.

https://stackoverflow.com/questions/21060073/dynamic-inheritance-in-python

**Review by gselzer — COMMENTED** [review]
Super close!

File: src/napari_imagej/types/widget_mappings.py (hunk 1 above)
Comment (gselzer) [reply src/napari_imagej/types/widget_mappings.py]: Added, and tested, in bc77b32

File: src/napari_imagej/widgets/parameter_widgets.py (hunk 3 above)
Comment (gselzer) [reply src/napari_imagej/widgets/parameter_widgets.py]: Implemented in bc77b32

**Pull request merged**